hm_1/tasks/task_02_05.py

```diff
@@ -0,0 +1,45 @@
+def detect_index(some_iterable_object, position):
+    if isinstance(some_iterable_object, dict):
```

Comment:
When the task says iterable it means any iterable and you may use only the iteration part.  You don't need to go deeper and handle some special cases of iterable.  Please remove this check from here.